lib/ts/TextView.cc

```diff
@@ -96,18 +95,30 @@ ts::svtoi(TextView src, TextView *out, int base)
   if (out) {
     out->clear();
   }
-  if (!(1 < base && base <= 36)) {
+  if (!(0 <= base && base <= 36)) {
     return 0;
   }
-  if (src.ltrim_if(&isspace)) {
+  if (src.ltrim_if(&isspace) && src) {
     const char *start = src.data();
     int8_t v;
     bool neg = false;
     if ('-' == *src) {
       ++src;
       neg = true;
     }
-    while (src.size() && (-1 != (v = convert[static_cast<unsigned char>(*src)]))) {
+    // If base is 0, it wasn't specified - check for standard base prefixes
+    if (0 == base) {
+      base = 10;
+      if ('0' == *src) {
+        ++src;
+        base = 8;
+        if (src && ('x' == *src || 'X' == *src)) {
```

Comment:
Shouldn't there be a test for this?

Reply:
There is. See lines 208-209 in `lib/ts/unit-tests/test_TextView.cc`.